demisto_sdk/commands/common/tools.py

```diff
@@ -225,60 +226,72 @@ def get_remote_file(
         The file content in the required format.
 
     """
-    github_config = GithubContentConfig(github_repo)
-    # 'origin/' prefix is used to compared with remote branches but it is not a part of the github url.
-    github_tag = tag.replace('origin/', '').replace('demisto/', '')
+    git_config = GitContentConfig(github_repo)
+    if git_config.GITLAB_ID is not None:
+        full_file_path_quote_plus = urllib.parse.quote_plus(full_file_path)
+        git_path = urljoin(git_config.BASE_RAW_GITLAB_LINK, 'files', full_file_path_quote_plus, 'raw')
+        tag = tag.replace('origin/', '')
+    else:  # github
+        # 'origin/' prefix is used to compared with remote branches but it is not a part of the github url.
+        tag = tag.replace('origin/', '').replace('demisto/', '')
+        git_path = urljoin(git_config.CONTENT_GITHUB_LINK, tag, full_file_path)
+
     local_content = '{}'
 
-    github_path = urljoin(github_config.CONTENT_GITHUB_LINK, github_tag, full_file_path)
-    github_token: Optional[str] = None
+    git_token: Optional[str] = None
     try:
         external_repo = is_external_repository()
         if external_repo:
-            githhub_config = GithubContentConfig()
-            github_token = githhub_config.Credentials.TOKEN
-            if github_token:
-                res = requests.get(github_path, verify=False, timeout=10, headers={
-                    'Authorization': f"Bearer {github_token}",
-                    'Accept': f'application/vnd.github.VERSION.raw',
-                })  # Sometime we need headers
-                if not res.ok:  # sometime we need param token
-                    res = requests.get(
-                        github_path,
-                        verify=False,
-                        timeout=10,
-                        params={'token': github_token}
-                    )
+            git_token = git_config.Credentials.TOKEN
+            if git_token:
+                if git_config.GITLAB_ID is not None:
+                    res = requests.get(git_path,
+                                       params={'ref': tag},
+                                       headers={'PRIVATE-TOKEN': git_token},
+                                       verify=False)
+                else:
+                    res = requests.get(git_path, verify=False, timeout=10, headers={
+                        'Authorization': f"Bearer {git_token}",
+                        'Accept': f'application/vnd.github.VERSION.raw',
+                    })  # Sometime we need headers
+                    if not res.ok:  # sometime we need param token
+                        res = requests.get(
+                            git_path,
+                            verify=False,
+                            timeout=10,
+                            params={'token': git_token}
+                        )
```

Comment:
This logic works both for GitLab and GitHub?

Reply:
I'm doing a different request for gitlab and github